cmds/wget/wget.go

```diff
@@ -48,13 +61,30 @@ func usage() {
 }
 
 func main() {
-	flag.Parse()
-	if flag.NArg() != 1 {
+	if flag.Parse(); flag.NArg() != 1 {
 		usage()
 	}
 
-	url := flag.Arg(0)
-	if err := wget(url, os.Stdout); err != nil {
-		log.Fatalf("%v\n", err)
+	argURL := flag.Arg(0)
+	if argURL == "" {
+		log.Fatalln("Empty URL")
+	}
+
+	url, err := url.Parse(argURL)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	fileName := "index.html"
```

Comment:
fileName := *outPath
if fileName = "" {
fileName = "index.html"
if url.Path != "" && etc. etc.
}

then you can drop that else down there. Just a bit simpler. And you could even make outPath a flag.Var:
flag.StringVar(&outPath, etc)
and skip having a new variable. 
if outPath == "" {
outPath = "index.html"
etc. etc.
}

Reply:
Is index.html really the default behavior of wget? You can't really know what the name of that file is as the client?

Reply:
@hugelgupf wget gets the filename from the last path component from the URL.
```
$ wget 'http://example.com/index.html?randomarg=123&arg2=xxx'
--2017-12-17 22:18:51--  http://example.com/index.html?randomarg=123&arg2=xxx
Resolving example.com (example.com)... 93.184.216.34, 2606:2800:220:1:248:1893:25c8:1946
Connecting to example.com (example.com)|93.184.216.34|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1270 (1.2K) [text/html]
Saving to: 'index.html?randomarg=123&arg2=xxx'

index.html?randomarg= 100%[========================>]   1.24K  --.-KB/s    in 0s      

2017-12-17 22:18:52 (83.0 MB/s) - 'index.html?randomarg=123&arg2=xxx' saved [1270/1270]
```
Alough this is the mainstream behaviour I don't think it's really correct, hence my idea to use url.Parse to get the filename without the arguments.

And yes, index.html is the default filename where it can't be detected from the URL, eg. trailing slash or no path
```
$ wget http://example.com
--2017-12-17 22:13:38--  http://example.com/
Resolving example.com (example.com)... 93.184.216.34, 2606:2800:220:1:248:1893:25c8:1946
Connecting to example.com (example.com)|93.184.216.34|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1270 (1.2K) [text/html]
Saving to: 'index.html'

index.html          100%[===================>]   1.24K  --.-KB/s    in 0s      

2017-12-17 22:13:39 (123 MB/s) - 'index.html' saved [1270/1270]
```

Reply:
@hugelgupf Well, there is a way to obtain the filename in some circumstances looking at the `Content-Disposition` header:
```
Content-Disposition: attachment; filename="filename.jpg"
```
but it seems that currently both wget and curl ignore it